### PR TITLE
Connect api profile reuse form

### PR DESCRIPTION
**Pull request by brylie**
Related to PR #239 

# Changes
- re-use Organization APIs form
- restructure some code
- cleanup
- lint Organizations and Organization APIs component code
  - this was agreed with @mauriciovieira, based on his [work with manual linting of the entire project](https://github.com/Digipalvelutehdas/APIKA/issues/236#issuecomment-273626743)

### CODE REVIEW

**Comment by marla-singer** [comment]
@brylie Reviewing

**Comment by marla-singer** [comment]
@brylie 

1. Specify right label and help text for modal form
joxi_screenshot_1484826314834

2. These is showing incorrect message when modal form is closed

File: organizations/client/profile/connect_api/autoform.js [hunk 1]
```diff
@@ -1,22 +0,0 @@
-import { AutoForm } from 'meteor/aldeed:autoform';
-import { TAPi18n } from 'meteor/tap:i18n';
-import { sAlert } from 'meteor/juliancwirko:s-alert';
-import { Modal } from 'meteor/peppelg:bootstrap-3-modal';
-
-AutoForm.hooks({
-  connectionApiForm: {
-    onSuccess () {
-      // Create & show message about successfully saving
-      const message = TAPi18n.__('connectionApiForm_successText');
```
Comment (marla-singer) [inline organizations/client/profile/connect_api/autoform.js]: Don't forget delete useless tags

File: organizations/client/profile/connect_api/connect_api.html [hunk 2]
```diff
@@ -13,25 +13,7 @@ <h1 class="modal-title">
         <div class="modal-body">
           {{# if dataIsReady }}
             {{# if unlinkedApis }}
-              {{# autoForm id="connectionApiForm" type="insert" collection=organizationApisCollection }}
-                <div class="form-group">
-                  <label>
-                    {{_ 'connectApiToOrganizationModal_labelText_selectApi' }}
```
Comment (marla-singer) [inline organizations/client/profile/connect_api/connect_api.html]: Don't forget to delete useless tags

File: organizations/client/profile/connect_api/connect_api.html [hunk 3]
```diff
@@ -13,25 +13,7 @@ <h1 class="modal-title">
...
-                  </label>
-                  <p class="help-text">
-                    {{_ 'connectApiToOrganizationModal_helpText' }}
-                  </p>
-                  <!-- Hide Organization ID field and autovalue -->
-                  {{> afFieldInput name="organizationId" value=organizationId type="hidden" }}
-                  {{> afFieldInput name="apiId" class="form-control" options=unlinkedApis }}
-                </div>
-                <button type="submit" class="btn btn-primary" id="connect-api-to-organization">
-                  {{_ 'connectApiToOrganizationModal_buttonText_addApi' }}
```
Comment (marla-singer) [inline organizations/client/profile/connect_api/connect_api.html]: Don't forget to delete useless tags

File: organizations/client/profile/connect_api/connect_api.html [hunk 4]
```diff
@@ -13,25 +13,7 @@ <h1 class="modal-title">
...
-                </button>
-                <button type="button" class="btn btn-default" data-dismiss="modal">
-                  {{_ 'connectApiToOrganizationModal_buttonText_cancel' }}
```
Comment (marla-singer) [inline organizations/client/profile/connect_api/connect_api.html]: Don't forget to delete useless tags

File: organizations/client/profile/connect_api/autoform.js (hunk 1 above)
Comment (brylie) [inline organizations/client/profile/connect_api/autoform.js]: Done.

File: organizations/client/profile/connect_api/connect_api.html (hunk 2 above)
Comment (brylie) [inline organizations/client/profile/connect_api/connect_api.html]: Done.

File: organizations/client/profile/connect_api/connect_api.html (hunk 3 above)
Comment (brylie) [inline organizations/client/profile/connect_api/connect_api.html]: Done.

File: organizations/client/profile/connect_api/connect_api.html (hunk 4 above)
Comment (brylie) [inline organizations/client/profile/connect_api/connect_api.html]: Done.

File: organizations/logo/collection/server/publications.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Meteor } from 'meteor/meteor';
 import { OrganizationLogo } from '/organizations/logo/collection/collection';
 
-Meteor.publish('allOrganizationLogo', function() {
+Meteor.publish('allOrganizationLogo', () => {
```
Comment (marla-singer) [inline organizations/logo/collection/server/publications.js]: Be careful. Don't use arrow-function on Publish method. See Meteor Docs
https://guide.meteor.com/data-loading.html#publications

**Comment by brylie** [comment]
> Specify right label and help text for modal form
 I updated the label localization in the i18n file. We probably don't need a help text for this form, but feel free to add it in your PR.

**Comment by brylie** [comment]
> These are showing an incorrect message when modal form is closed. Although the meaning of the message is correct.

I made the success message more generic, so it can apply in both scenarios.

File: organizations/logo/collection/server/publications.js (hunk 5 above)
Comment (marla-singer) [inline organizations/logo/collection/server/publications.js]: In this case, it doesn't matter but keep in mind

**Comment by marla-singer** [comment]
I think it's fine and is ready to merging

**Comment by brylie** [comment]
Cool beans!

**Comment by brylie** [comment]
Go ahead and merge, if you approve the changes.

**Pull request merged**